### PR TITLE
feat: read cloud config from secret 

### LINKED ISSUE #333: support read cloud config from secret
Hi * ,

I installing   plain k8s cluster and want to connect the k8s ckuster to azure account , 
there is any way to do it when the cloud config is not plain text.

BR Eliyahu

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind documentation
/kind feature
/kind flake

**What this PR does / why we need it**:

- read cloud config from secret 

- update the related deployment

- add the related doc

**Which issue(s) this PR fixes**:

Fixes #333 

**Special notes for your reviewer**:

**Release note**:
```
read cloud config from secret 
```

### CODE REVIEW

File: pkg/azuredisk/azure.go [hunk 1]
```diff
@@ -20,26 +20,52 @@ import (
...
+	az := &azure.Cloud{
+		KubeClient: kubeClient,
 	}
+	az.InitializeCloudFromSecret()
 
-	f, err := os.Open(credFile)
-	if err != nil {
-		klog.Errorf("Failed to load config from file: %s", credFile)
-		return nil, fmt.Errorf("Failed to load config from file: %s, cloud not get azure cloud provider", credFile)
+	if az.ResourceGroup == "" {
```
Comment (andyzhangx) [inline pkg/azuredisk/azure.go:41]: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/docs/cloud-provider-config.md#auth-configs
use `az.SubscriptionId` to tell whether it's already InitializeCloudFromSecret

File: charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node.yaml [hunk 2]
```diff
@@ -15,6 +15,7 @@ spec:
         app: csi-azuredisk-node
     spec:
       hostNetwork: true
+      serviceAccountName: csi-azuredisk-node-sa
```
Comment (andyzhangx) [inline charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node.yaml:18]: controller should also have access to the secret

Comment (ZeroMagic) [reply charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node.yaml:18]: has bound the role and service account of the controller in rbac-csi-azuredisk-secret.yaml

File: pkg/azuredisk/azure.go (hunk 1 above)
Comment (ZeroMagic) [reply pkg/azuredisk/azure.go]: done

File: pkg/azuredisk/azure.go [hunk 3]
```diff
@@ -20,26 +20,64 @@ import (
...
-	credFile, ok := os.LookupEnv("AZURE_CREDENTIAL_FILE")
-	if ok {
-		klog.V(2).Infof("AZURE_CREDENTIAL_FILE env var set as %v", credFile)
-	} else {
-		credFile = "/etc/kubernetes/azure.json"
-		klog.V(2).Infof("use default AZURE_CREDENTIAL_FILE env var: %v", credFile)
+func GetCloudProvider(kubeconfig string) (*azure.Cloud, error) {
+	klog.V(2).Infof("reading cloud config from secret")
+	kubeClient, err := getKubeClient(kubeconfig)
+	if err != nil && !os.IsNotExist(err) && err != rest.ErrNotInCluster {
```
Comment (ZeroMagic) [inline pkg/azuredisk/azure.go:34]: `!os.IsNotExist(err) && err != rest.ErrNotInCluster`
The two judgments ensure that we are able to test the driver in a local machine.

**Comment by ZeroMagic** [comment]
/hold cancel
All tests passed.

File: pkg/azuredisk/azure.go (hunk 3 above)
Comment (andyzhangx) [reply pkg/azuredisk/azure.go:34]: could you add a unit test for this? at least write a negative case for this functionality?

File: pkg/azuredisk/azure.go [hunk 4]
```diff
@@ -20,26 +20,64 @@ import (
...
 // GetCloudProvider get Azure Cloud Provider
-func GetCloudProvider() (*azure.Cloud, error) {
-	credFile, ok := os.LookupEnv("AZURE_CREDENTIAL_FILE")
-	if ok {
-		klog.V(2).Infof("AZURE_CREDENTIAL_FILE env var set as %v", credFile)
-	} else {
-		credFile = "/etc/kubernetes/azure.json"
-		klog.V(2).Infof("use default AZURE_CREDENTIAL_FILE env var: %v", credFile)
+func GetCloudProvider(kubeconfig string) (*azure.Cloud, error) {
+	klog.V(2).Infof("reading cloud config from secret")
```
Comment (andyzhangx) [inline pkg/azuredisk/azure.go:32]: move `klog.V(2).Infof("reading cloud config from secret")` to L40

File: pkg/azuredisk/azure.go [hunk 5]
```diff
@@ -19,27 +19,73 @@ package azuredisk
...
+	}
+
+	az := &azure.Cloud{}
+	if kubeClient != nil {
+		klog.V(2).Infof("reading cloud config from secret")
+		az.KubeClient = kubeClient
+		az.InitializeCloudFromSecret()
+	}
+
+	if reflect.DeepEqual(az, &azure.Cloud{}) {
```
Comment (ZeroMagic) [inline pkg/azuredisk/azure.go:50]: In order to follow the thought that we do not verify the azure.Cloud in func`GetCloudProvider()`, changed the judgment condition.  

When the user provides a wrong kubeconfig, kubeClient is still created. az will be initialize. Although it is invalid, it is created successfully, then return it.

File: pkg/azuredisk/azure.go (hunk 4 above)
Comment (ZeroMagic) [reply pkg/azuredisk/azure.go]: done

File: pkg/azuredisk/azure.go (hunk 3 above)
Comment (ZeroMagic) [reply pkg/azuredisk/azure.go:39]: done. Add some positive and negative cases.

**Comment by ZeroMagic** [comment]
All lines are hit. 
https://coveralls.io/builds/29935214/source?filename=pkg%2Fazuredisk%2Fazure.go
